experimental/hip/dynamic_symbols.h

```diff
@@ -30,13 +30,16 @@ typedef struct iree_hal_hip_dynamic_symbols_t {
   iree_dynamic_library_t* dylib;
 
   // Concrete HIP symbols defined by including the `dynamic_symbol_tables.h`.
-#define IREE_HIP_PFN_DECL(hipSymbolName, ...) \
+#define REQUIRED_IREE_HIP_PFN_DECL(hipSymbolName, ...) \
```

Comment:
prefer IREE_* as a prefix, so IREE_HAL_HIP_OPTIONAL_PFN_DECL and IREE_HAL_HIP_REQUIRED_PFN_DECL - macro names should be like namespaces - anything starting with IREE_ should not ever conflict with anything not IREE, and anything IREE_HAL_HIP_ should not conflict with anything not related to IREE + HIP HAL.